internal/provider/resource_tfe_organization_run_task.go

```diff
@@ -194,24 +302,48 @@ func (r *resourceOrgRunTask) Create(ctx context.Context, req resource.CreateRequ
 		Name:        plan.Name.ValueString(),
 		URL:         plan.URL.ValueString(),
 		Category:    plan.Category.ValueString(),
-		HMACKey:     plan.HMACKey.ValueStringPointer(),
 		Enabled:     plan.Enabled.ValueBoolPointer(),
 		Description: plan.Description.ValueStringPointer(),
 	}
 
+	if !config.HMACKeyWO.IsNull() {
+		options.HMACKey = config.HMACKeyWO.ValueStringPointer()
+	} else {
+		options.HMACKey = plan.HMACKey.ValueStringPointer()
+	}
+
 	tflog.Debug(ctx, fmt.Sprintf("Create task %s for organization: %s", options.Name, organization))
 	task, err := r.config.Client.RunTasks.Create(ctx, organization, options)
 	if err != nil {
 		resp.Diagnostics.AddError("Unable to create organization task", err.Error())
 		return
 	}
 
-	result := modelFromTFEOrganizationRunTask(task, plan.HMACKey)
+	result := modelFromTFEOrganizationRunTask(task, plan.HMACKey, !config.HMACKeyWO.IsNull())
+	fmt.Printf("In Create method")
+
+	fmt.Printf("--- config.HMACKeyWO: %+v\n", config.HMACKeyWO)
+	if !config.HMACKeyWO.IsNull() {
+		// Use the resource's private state to store secure hashes of write-only argument values, the provider during planmodify will use the hash to determine if a write-only argument value has changed in later Terraform runs.
+		hashedValue := generateSHA256Hash(config.HMACKeyWO.ValueString())
+		diags := resp.Private.SetKey(ctx, "hmac_key_wo", fmt.Appendf(nil, `"%s"`, hashedValue))
+		resp.Diagnostics.Append(diags...)
+	} else {
+		// if the key is not configured as write-only, then remove HMACKeyWO key from private state. Setting a key with an empty byte slice is interpreted by the framework as a request to remove the key from the ProviderData map.
+		diags := resp.Private.SetKey(ctx, "hmac_key_wo", []byte(""))
+		resp.Diagnostics.Append(diags...)
+	}
 
 	// Save data into Terraform state
 	resp.Diagnostics.Append(resp.State.Set(ctx, &result)...)
 }
 
+func generateSHA256Hash(data string) string {
+	hasher := sha256.New()
+	hasher.Write([]byte(data))
+	return hex.EncodeToString(hasher.Sum(nil))
+}
```

Comment:
This one is easier. @uturunku1 already moved this method to a shared location, so you can just delete this copy.